persist/fs/commitlog/writer.go

```diff
@@ -154,19 +157,38 @@ func (w *writer) isOpen() bool {
 	return w.chunkWriter.fd != nil
 }
 
+func (w *writer) logEntryIndex(seriesUniqueIndex uint64) uint64 {
+	// Special case the zeroth value as we compare the value
+	// at the entry of the seen cache with the index being put
+	// at the location - since all entries in the seen cache when
+	// not set are zero the value being used to compare whether
+	// this result has been seen or not cannot be zero itself
+	return seriesUniqueIndex + 1
```

Comment:
Ah makes sense. Maybe re-word the comment to clarify why we're doing this, it took me a minute to figure out what you meant. Something like: `This value is used as the series' unique index in the seenCache. The seenCache is zero initialised, so we add one to ensure the value we set in the slice is non-zero. This lets us distinguish un-initialised v initialised seenCache entries.`